ls1intum/Artemis
# `Plagiarism checks`: Fix empty clean up dialog

### Checklist
#### General





#### Client


### Motivation and Context


This PR aims to fix #8528. When running a plagiarism check on a text-based exercise, there is no chart showing up + the clean up dialog is bugged. This is due to the sent data being faulty, and the client not handling it correctly.


### Description

The client is normally laid out to handle a plagiarismResult Object with a similarity distribution array of length 10. However, somehow the backend is sending an array of length 100 when checking for plagiarisms on Text-Based Exercises. This PR implements an algorithm that takes the bigger array and reduces the values inbetween (indexes Y0 - Y9) to one value and therefore transforms the array to length 10 to be passed to the ngx chart. By getting rid of the ngx chart instantiation errors, the modal is then able to be displayed properly.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Text based Programming Exercise (no modeling exercise)


1. Create course
2. Create exercise
3. Submit two solutions as student
4. As instructor, run plagiarism check
5. Check whether the chart for similarity distribution is displayed properly and the "Clean Up" 

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test
